doc/unordered/hash_traits.adoc

```diff
@@ -32,9 +32,14 @@ large changes in the returned hash code &#8212;ideally, flipping one bit in the
 the input value results in each bit of the hash code flipping with probability 50%. Approaching
 this property is critical for the proper behavior of open-addressing hash containers.
 
-`hash_is_avalanching<Hash>::value` is `true` if `Hash::is_avalanching` is a valid type,
-and `false` otherwise.
-Users can then declare a hash function `Hash` as avalanching either by embedding an `is_avalanching` typedef
+`hash_is_avalanching<Hash>::value` is:
+
+ * `false` if `Hash::is_avalanching` is not present,
```

Comment:
What do we want when `is_avalanching` is present, but is not a type? `false` or a compile error? The latter, probably.

Reply:
We can add a specialization like this:
```cpp
template<typename Hash>
struct hash_is_avalanching_impl<
  Hash,
  typename std::enable_if<((void)Hash::is_avalanching,true)>::type
>{};
```

Reply:
Let's just document it as `false` and be done with it. It's way too convoluted otherwise.

Reply:
Maybe we _should_ make it a compile-time error to protect users from:
```cpp
struct my_hash
{
  static constexpr bool is_avalanching=true;
};
```

Reply:
That's exactly what I thought, but if it's too much of a hassle, we shouldn't go out of our way to try and catch that.

Reply:
It's not a hassle, the change proposed above does indeed do the trick.

Reply:
Fwiw, if Joaquin doesn't mind putting in the work, I also think making it a compiler error would be fantastic. As a user of Unordered, this is exacly the behavior I'd want (i.e. saving me from myself).